src/bson.rs

```diff
@@ -320,7 +320,7 @@ impl Bson {
         match j {
             &Value::Number(ref x) =>
                 x.as_i64().map(Bson::from)
-                .or(x.as_u64().map(Bson::from))
+                .or(x.as_f64().map(Bson::from))
```

Comment:
One historical bug, we should use `.or_else(|_| x.as_f64().map(Bson::from)` to prevent unnecessary method calls.